### DISCUSSION TITLE
LEGO as Server library or smth Similar

### DISCUSSION THREAD

**Discussion by DaspawnW** [go-acme/lego#2087] [Q&A]
I really like the lego project. It's a very easy way to implement your own Provider :-) Thanks for this project.

I'm right now interrested on the oposite side, so a good way to implement a ACME compatible Server where the DNS & Http Challenge is already handled and I only have to implement the methods to take the CSR and sign it. So very similar to the Provider approach you already have in place, but as said for the oposite side.

Are you aware of any Project like LEGO in golang that implements the ACME Server Side?

**Comment by ldez** [comment]
- https://github.com/letsencrypt/pebble
- https://github.com/letsencrypt/boulder
- https://github.com/smallstep/certificates
